include/core/CCompressedLfuCache.h

```diff
@@ -453,15 +469,17 @@ class CCompressedLfuCache {
     }
 
     bool full(std::size_t itemMemoryUsage) const {
-        double memory{static_cast<double>(this->unguardedMemoryUsage() + itemMemoryUsage)};
+        std::size_t memory{this->unguardedMemoryUsage() + itemMemoryUsage +
+                           sizeof(typename TCompressedKeyCacheItemUMap::value_type) +
+                           sizeof(typename TCacheItemStatsSet::value_type)};
```

Comment:
We also get 2X sizeof(std::size_t) per node, which explains the overshoot in the unit tests.